### PR TITLE
Refactoring (+ space settings/presets)

### PR DESCRIPTION
**Pull request by skddc**
When I intended to continue hacking on the settings branch yesterday, it got a little bit out of hand and I ended up refactoring most of the `smt` service and splitting out IRC functions to its own `sockethub-irc` service in preparation for a `sockethub-xmpp` and potential other adapters. In theory the AS objects should look mostly the same in both IRC and XMPP, but this way it's much easier to hack forward fast, and then later compare which parts can be unified, without breaking either adapter while hacking on the other one.

Anyway, this doesn't introduce major changes, so maybe just scan the code and smoke-test if stuff still works. I actually found a bug which entered `master` recently, and also fixed that just now.

Edit: turns out we were still testing on node 4, and that didn't likesome new syntax outside of strict mode. So I also changed it to test node 6 and require that for devs, too.

### CODE REVIEW

File: tests/unit/services/sockethub-irc-test.js [hunk 1]
```diff
@@ -0,0 +1,17 @@
+import { moduleFor, test } from 'ember-qunit';
+
+moduleFor('service:sockethub-irc', 'Unit | Service | sockethub irc', {
+  needs: [
+    'service:ajax',
+    'service:logger',
+    'service:remotestorage',
+    'service:coms'
```
Comment (galfert) [inline tests/unit/services/sockethub-irc-test.js]: See comment above, `coms` is not needed.

File: app/services/sockethub-irc.js [hunk 2]
```diff
@@ -0,0 +1,180 @@
...
+} = Ember;
+
+/**
+ * This service provides helpers for SocketHub IRC communications
+ * @module hyperchannel/services/sockethub-irc
+ */
+export default Ember.Service.extend({
+
+  logger: service(),
+  coms: service(),
```
Comment (skddc) [inline app/services/sockethub-irc.js]: This service's unit test actually makes sure it has access to coms. When we move more functions here it'll be needed I think.

**Comment by galfert** [comment]
Did some smoke testing and couldn't find any problems.

**Comment by skddc** [comment]
Cool, thanks.

**Comment by galfert** [comment]
Btw, I started to add modals, so we can use them for the preset config. Will try to finish that tomorrow.

**Comment by skddc** [comment]
Oh, nice!

**Pull request merged**